src/stdlib_io_disp.fypp

```diff
@@ -0,0 +1,125 @@
+#:include "common.fypp"
+#:set RIL_KINDS_TYPES = REAL_KINDS_TYPES + INT_KINDS_TYPES + LOG_KINDS_TYPES
+
+submodule (stdlib_io) stdlib_io_disp
+
+    use, non_intrinsic :: stdlib_strings, only: format_string
+    implicit none
+    character(len=*), parameter :: fmt_r = '(*(g11.4, 1x))'
+    character(len=*), parameter :: fmt_c = '(*(g23.4, 1x))'
+    
+contains
+    
+    #:for kind, type in RIL_KINDS_TYPES
+    module procedure disp_0_${type[0]}$${kind}$
+        if(present(string)) print *, trim(string)
+        print fmt_r, val
+    end procedure disp_0_${type[0]}$${kind}$
+
+    module procedure disp_1_${type[0]}$${kind}$
+        integer :: i, m
+        m = size(val)
+        if(present(string)) print *, trim(string)
+        print fmt_r, (val(i), i=1,m)
+    end procedure disp_1_${type[0]}$${kind}$
+
+    module procedure disp_2_${type[0]}$${kind}$
+        integer :: i, j, m, n
+        m = size(val, 1)
+        n = size(val, 2)
+        if(present(string)) print *, trim(string)
+        do i = 1, m
+            print fmt_r, (val(i,j), j=1, n)
+        end do
+    end procedure disp_2_${type[0]}$${kind}$
+
+    module procedure disp_3_${type[0]}$${kind}$
+        integer :: i, dim1, dim2, dim3
+        dim1 = size(val, 1)
+        dim2 = size(val, 2)
+        dim3 = size(val, 3)
+        if(present(string)) print *, trim(string)
+        if(dim == 1) then
+            do i = 1, dim1
+                print *, 'Slice ('//format_string(i)//',:,:):'
```

Comment:
You can use `to_string` from `stdlib_ascii` here, it usually faster than internal IO and yields the same result

Reply:
Great suggestion! I am not so familiar with the code inside `stdlib`, thank you for your reminder. This is also the reason why I recommend `stdlib` to quickly support `fpm`(see https://github.com/fortran-lang/stdlib/pull/437), so that users like me can use `stdlib` directly in `fpm` and become familiar with it.